examples/jsm/objects/BatchedMesh.js

```diff
@@ -0,0 +1,461 @@
+import {
+	BufferAttribute,
+	BufferGeometry,
+	DataTexture,
+	FloatType,
+	MathUtils,
+	Matrix4,
+	Mesh,
+	RGBAFormat
+} from 'three';
+
+const _identityMatrix = new Matrix4();
+const _zeroMatrix = new Matrix4().set(
+	0, 0, 0, 0,
+	0, 0, 0, 0,
+	0, 0, 0, 0,
+	0, 0, 0, 0
+);
+
+// Arbitary numbers
+const DEFAULT_MAX_GEOMETRY_COUNT = 256;
+const DEFAULT_MAX_VERTEX_COUNT = 1024 * 256;
+const DEFAULT_MAX_INDEX_COUNT = 1024 * 256;
+
+// Custom shaders
+const batchingParsVertex = `
+#ifdef BATCHING
+	attribute float id;
+	uniform highp sampler2D batchingTexture;
+	uniform int batchingTextureSize;
+	mat4 getBatchingMatrix( const in float i ) {
+		float j = i * 4.0;
+		float x = mod( j, float( batchingTextureSize ) );
+		float y = floor( j / float( batchingTextureSize ) );
+		float dx = 1.0 / float( batchingTextureSize );
+		float dy = 1.0 / float( batchingTextureSize );
+		y = dy * ( y + 0.5 );
+		vec4 v1 = texture2D( batchingTexture, vec2( dx * ( x + 0.5 ), y ) );
+		vec4 v2 = texture2D( batchingTexture, vec2( dx * ( x + 1.5 ), y ) );
+		vec4 v3 = texture2D( batchingTexture, vec2( dx * ( x + 2.5 ), y ) );
+		vec4 v4 = texture2D( batchingTexture, vec2( dx * ( x + 3.5 ), y ) );
+		mat4 bone = mat4( v1, v2, v3, v4 );
+		return bone;
+	}
+#endif
+`;
+
+const batchingbaseVertex = `
+#ifdef BATCHING
+	mat4 batchingMatrix = getBatchingMatrix( id );
+#endif
+`;
+
+const batchingnormalVertex = `
+#ifdef BATCHING
+	objectNormal = vec4( batchingMatrix * vec4( objectNormal, 0.0 ) ).xyz;
+	#ifdef USE_TANGENT
+		objectTangent = vec4( batchingMatrix * vec4( objectTangent, 0.0 ) ).xyz;
+	#endif
+#endif
+`;
+
+const batchingVertex = `
+#ifdef BATCHING
+	transformed = ( batchingMatrix * vec4( transformed, 1.0 ) ).xyz;
+#endif
+`;
+
+// @TODO: SkinnedMesh support?
+// @TODO: Move into the core. Can be optimized more with WEBGL_multi_draw.
+
+class BatchedMesh extends Mesh {
+
+	constructor( material, maxGeometryCount = DEFAULT_MAX_GEOMETRY_COUNT,
+		maxVertexCount = DEFAULT_MAX_VERTEX_COUNT, maxIndexCount = DEFAULT_MAX_INDEX_COUNT ) {
```

Comment:
I think maxGeometryCount and maxVertexCount are probably two parameters that users really need to provide an intentional value for. We don't give a default count for InstancedMesh either. I'd be fine with material or maxIndexCount maybe having default values though. How about:

```js
constructor( maxGeometryCount, maxVertexCount, maxIndexCount = maxVertexCount * 2, material = new MeshBasicMaterial() );
```

Reply:
Sounds good to me. Updated the constructor parameters. Default material is defined in `Mesh` (super) constructor, so I don't define it here.